### PR TITLE
Get result from applied arm template

### PR DESCRIPTION
**Pull request by kargakis**
/hold

### CODE REVIEW

File: pkg/util/azureclient/resources.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 type DeploymentsClient interface {
 	CreateOrUpdate(ctx context.Context, resourceGroupName string, deploymentName string, parameters resources.Deployment) (result resources.DeploymentsCreateOrUpdateFuture, err error)
 	Client
+	DeploymentClient() resources.DeploymentsClient
```
Comment (kargakis) [inline pkg/util/azureclient/resources.go]: @mjudeikis @asalkeld this one looks like a limitation of our sdk interfaces.

Comment (mjudeikis) [reply pkg/util/azureclient/resources.go]: it is. But this way we can mock any interfaces with mock data for unti tests

File: pkg/log/log.go [hunk 2]
```diff
@@ -59,6 +59,11 @@ func Fatalf(format string, args ...interface{}) {
 	logger.log.Fatalf(format, args...)
 }
 
+// Warn logs at warn level
+func Warn(format string, args ...interface{}) {
```
Comment (kargakis) [inline pkg/log/log.go]: Honestly I am sick of maintaining all these shims already

**Comment by kargakis** [comment]
> @kargakis remove hold?

I would like to see https://github.com/openshift/openshift-azure/issues/615#issuecomment-433337154 answered first.

File: pkg/log/log.go (hunk 2 above)
Comment (charlesakalugwu) [inline pkg/log/log.go]: this should be `Warnf` ...however we probably want both versions `Warn` and `Warnf`

Comment (kargakis) [reply pkg/log/log.go]: Yes, we want both. I should get rid of the formatting here since the correct use is Warn. I am adding Warnf in a different PR.

**Pull request merged**